src/events/browser-network-status-listener.spec.ts

```diff
@@ -0,0 +1,140 @@
+import BrowserNetworkStatusListener from './browser-network-status-listener';
+
+describe('BrowserNetworkStatusListener', () => {
+  let originalNavigator: Navigator;
+  let originalWindow: Window;
+
+  beforeEach(() => {
+    // Save original references
+    originalNavigator = global.navigator;
+    originalWindow = global.window;
+
+    // Mock `navigator.onLine`
+    Object.defineProperty(global, 'navigator', {
+      value: { onLine: true },
+      writable: true,
+    });
+
+    const listeners: Map<string, (offline: boolean) => void> = new Map();
+    Object.defineProperty(global, 'window', {
+      value: {
+        addEventListener: (evt: string, fn: () => void) => {
+          listeners.set(evt, fn);
+        },
+        removeEventListener: () => null,
+        dispatchEvent: (event: Event) => {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          const listener = listeners.get(event.type)!;
+          listener(event.type === 'offline');
+        },
+      },
+      writable: true,
+    });
+  });
+
+  afterEach(() => {
+    // Restore original references
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore test code
+    // noinspection JSConstantReassignment
+    global.navigator = originalNavigator;
+    // noinspection JSConstantReassignment
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore test code
+    // noinspection JSConstantReassignment
+    global.window = originalWindow;
+
+    jest.clearAllMocks();
+  });
+
+  test('throws an error if instantiated outside a browser environment', () => {
+    Object.defineProperty(global, 'window', { value: undefined });
+
+    expect(() => new BrowserNetworkStatusListener()).toThrow(
+      'BrowserNetworkStatusListener can only be used in a browser environment',
+    );
+  });
+
+  test('correctly initializes offline state based on navigator.onLine', () => {
+    // Online state
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore test code
+    // noinspection JSConstantReassignment
+    navigator.onLine = true;
```

Comment:
woah cool I didn't even know browsers had this 
![image](https://github.com/user-attachments/assets/2e91e2e2-85e0-4842-9724-b4ccd492cbe0)

Reply:
🎉 despite the awful `onLine` casing 🤮